openqaoa/optimizers/training_vqa.py

```diff
@@ -646,5 +648,160 @@ def optimize(self):
+                                  'pennylane_rmsprop', 'pennylane_rotosolve', 'pennylane_spsa']
+
+    def __init__(self,
+                 vqa_object: Type[VQABaseBackend],
+                 variational_params: Type[QAOAVariationalBaseParams],
+                 optimizer_dict: dict):
+
+        super().__init__(vqa_object, variational_params, optimizer_dict)
+
+        self.vqa_object = vqa_object
+        self._validate_and_set_params(optimizer_dict)
+
+    def _validate_and_set_params(self, optimizer_dict):
+        """
+        Verify that the specified arguments are valid for the particular optimizer.
+        """
+
+        if self.method not in PennyLaneOptimizer.PENNYLANE_OPTIMIZERS:
+            raise ValueError(
+                f"Please choose from the supported methods: {PennyLaneOptimizer.PENNYLANE_OPTIMIZERS}")
+
+        jac = optimizer_dict.get('jac', None)
+        jac_options = optimizer_dict.get('jac_options', None)
+
+        if jac is None or not isinstance(jac, (Callable, str)):
+            raise ValueError(
+                "Please specify either a string or provide callable gradient in order to use gradient based methods")
+        else:
+            if isinstance(jac, str):
+                self.jac = derivative(self.vqa_object, self.variational_params, self.log, 'gradient', jac, jac_options)
+            else:
+                self.jac = jac
+
+        constraints = optimizer_dict.get('constraints', ())
+        if constraints == () or isinstance(constraints, LinearConstraint) or isinstance(constraints, NonlinearConstraint):
+            self.constraints = constraints
+        else:
+            raise ValueError(
+                f"Constraints for Scipy optimization should be of type {LinearConstraint} or {NonlinearConstraint}")
+
+        bounds = optimizer_dict.get('bounds', None)
+        if bounds is None or isinstance(bounds, Bounds):
+            self.bounds = bounds
+        else:
+            raise ValueError(
+                f"Bounds for Scipy optimization should be of type {Bounds}")
+
+        self.options = optimizer_dict.get('optimizer_options', {})
+        self.options["maxiter"] = optimizer_dict.get('maxiter', None)
+        if optimizer_dict.get('maxfev') is not None:
+            self.options["maxfev"] = optimizer_dict.get('maxfev', None)
+
+        self.tol = optimizer_dict.get('tol', None)
+
+        return self
+
+    def __repr__(self):
+        """
+        Overview of the instantiated optimier/trainer.
+        """
+        maxiter = self.options["maxiter"]
+        string = f"Optimizer for VQA of type: {type(self.vqa).__base__.__name__} \n"
+        string += f"Backend: {type(self.vqa).__name__} \n"
+        string += f"Method: {str(self.method).upper()} with Max Iterations: {maxiter}\n"
+
+        return string
+
+    def optimize(self):
+        '''
+        Main method which implements the optimization process using ``scipy.minimize``.
+
+        Returns
+        -------
+        : 
```

Comment:
Is this docstring correct?

Reply:
Do you mean the format? The format is copied from the CustomScipyGradientOptimizer. Should I change both?